File: Typescript/blockchain/token.ts
class Token {
    public name: string;
    public symbol: string;
    public totalSupply: number;
    public balance: { [key: string]: number };

    constructor(name: string, symbol: string, totalSupply: number) {
        this.name = name;
        this.symbol = symbol;
        this.totalSupply = totalSupply;
        this.balance = {};
    }

    mint(account: string, amount: number): void {
        if (this.balance[account]) {
            this.balance[account] += amount;
        } else {
            this.balance[account] = amount;
        }
    }

    transfer(sender: string, receiver: string, amount: number): boolean {
        if (this.balance[sender] && this.balance[sender] >= amount) {
            this.balance[sender] -= amount;
            this.balance[receiver] = this.balance[receiver] ? this.balance[receiver] + amount : amount;
            return true;
        }
        return false;
    }
}

// Usage example:
const myToken = new Token("MyToken", "MTK", 1000000);
myToken.mint("Address1", 500);
myToken.transfer("Address1", "Address2", 200);

File: PyScripts/blockchain/token.py
class Token:
    def __init__(self, name, symbol, total_supply):
        self.name = name
        self.symbol = symbol
        self.total_supply = total_supply
        self.balance = {}

    def mint(self, account, amount):
        if account in self.balance:
            self.balance[account] += amount
        else:
            self.balance[account] = amount

    def transfer(self, sender, receiver, amount):
        if sender in self.balance and self.balance[sender] >= amount:
            self.balance[sender] -= amount
            if receiver in self.balance:
                self.balance[receiver] += amount
            else:
                self.balance[receiver] = amount
            return True
        return False

# Usage example:
my_token = Token("MyToken", "MTK", 1000000)
my_token.mint("Address1", 500)
my_token.transfer("Address1", "Address2", 200)
